edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.CloudProxy/NestedDeviceScopeApiClient.cs

```diff
@@ -160,6 +160,15 @@ async Task<ScopeResult> GetIdentityOnBehalfOf(Uri uri, string deviceId, Option<s
             {
                 // Get the auth-chain for the target device
                 Option<string> maybeAuthChain = await this.serviceIdentityHierarchy.GetEdgeAuthChain(deviceId);
+
+                if (!maybeAuthChain.HasValue)
+                {
+                    // We might not have the target identity yet (e.g. it just got added).
+                    // In this case, we can still try to make the call as ourself, in case
+                    // the missing identity is one of our immediate children.
+                    maybeAuthChain = await this.serviceIdentityHierarchy.GetAuthChain(this.actorEdgeDeviceId);
```

Comment:
Who will make the call if the target is one of our immediate children? The EdgeHub itself? Do we will use this codepath for that flow?

Reply:
the authenticator makes the call in that scenario, if we fail to find the auth target, we'll attempt to refresh the target identity and try again.  This block handles that by mirroring our legacy capability (GetDeviceAndModuleInDeviceScope).